### PR TITLE
Adding website documentation for private state management

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #449

### CODE REVIEW

File: website/docs/plugin/framework/resources/private-state.mdx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+read, and update operations and the ability to read this data during delete, planning, read, and update operations.
+
+## Accessing Private State Data
+
+Private state data can be read from a [privatestate.ProviderData](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/internal/privatestate#ProviderData) 
+type in the `Private` field present in the _request_ that is passed into:
+
+* `Delete`, `Read` and `Update` functions that form part of the [resource.Resource](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#Resource) interface.
+* `ModifyPlan` function that forms part of the [resource.ResourceWithModifyPlan](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithModifyPlan) interface.
+* `Modify` function that forms part of the [tfsdk.AttributePlanModifier](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#AttributePlanModifier) interface.
```
Comment (bflad) [inline website/docs/plugin/framework/resources/private-state.mdx:36]: I'd recommend splitting these out and linking directly to the fields themselves:

```suggestion
| Resource Operation | Private State Data |
| --- | --- |
| Delete | [resource.DeleteRequest.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#DeleteRequest.Private) |
| Plan Modification ([resource.ResourceWithModifyPlan](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithModifyPlan)) | [resource.ModifyPlanRequest.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ModifyPlanRequest.Private) |
| Plan Modification ([tfsdk.AttributePlanModifier](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#AttributePlanModifier)) | [tfsdk.ModifyAttributePlanRequest.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#ModifyAttributePlanRequest.Private) |
| Read | [resource.ReadRequest.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ReadRequest.Private) |
| Update | [resource.UpdateRequest.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#UpdateRequest.Private)
```

File: website/docs/plugin/framework/resources/private-state.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Private state data can be saved to a [privatestate.ProviderData](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/internal/privatestate#ProviderData) 
+type in the `Private` field present in the _response_ that is returned from:
+
+* `Create`, `Read` and `Update` functions that form part of the [resource.Resource](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#Resource) interface.
+* `ImportState` function that forms part of the [resource.ResourceWithImportState](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithImportState) interface.
+* `ModifyPlan` function that forms part of the [resource.ResourceWithModifyPlan](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithModifyPlan) interface.
+* `Modify` function that forms part of the [tfsdk.AttributePlanModifier](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#AttributePlanModifier) interface.
```
Comment (bflad) [inline website/docs/plugin/framework/resources/private-state.mdx:44]: Similarly:

```suggestion
| Resource Operation | Private State Data |
| --- | --- |
| Create | [resource.CreateResponse.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#CreateResponse.Private) |
| Import | [resource.ImportStateResponse.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ImportStateResponse.Private) |
| Plan Modification ([resource.ResourceWithModifyPlan](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ResourceWithModifyPlan)) | [resource.ModifyPlanResponse.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ModifyPlanResponse.Private) |
| Plan Modification ([tfsdk.AttributePlanModifier](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#AttributePlanModifier)) | [tfsdk.ModifyAttributePlanResponse.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#ModifyAttributePlanResponse.Private) |
| Read | [resource.ReadResponse.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#ReadResponse.Private) |
| Update | [resource.UpdateResponse.Private](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#UpdateResponse.Private)
```

File: website/docs/plugin/framework/resources/private-state.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	resp.Diagnostics.Append(diags...)
+	
+	if value != nil {
+		// value will be []byte.
+	    ...	
+    }
+}
+```
+
+If the key supplied is reserved for framework usage, an error diagnostic will be returned.
```
Comment (bflad) [inline website/docs/plugin/framework/resources/private-state.mdx:65]: We should discuss the reserved keys somewhere 😄

Comment (bendbennett) [reply website/docs/plugin/framework/resources/private-state.mdx]: @bflad I've added a few words into a `Reserved Keys` section.

**Pull request merged**